src/main/kotlin/net/lumalyte/lg/infrastructure/services/GuildServiceBukkit.kt

```
@@ -207,6 +207,17 @@ class GuildServiceBukkit(
             val bannerText = if (banner != null) "${banner.type.name} with patterns" else "cleared (default white banner)"
             logger.info("Guild $guildId banner set to '$bannerText' by $actorId")
             Bukkit.getPluginManager().callEvent(GuildBannerSetEvent(guildId, actorId))
+
+            // Notify bannerman renderer to re-render the new banner. Failures are non-fatal
+            // (Koin may be torn down during shutdown), but we log so operators can diagnose
+            // missing-refresh complaints instead of chasing ghost bug reports.
+            try {
+                val bannermanListeners = org.koin.java.KoinJavaComponent.getKoin()
+                    .get<net.lumalyte.lg.infrastructure.bukkit.bannerman.BannermanListeners>()
+                bannermanListeners.onGuildBannerChanged(guildId)
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)